src/Workspaces/Remote/Core/RemoteCallback.cs

```diff
@@ -118,20 +108,24 @@ private bool ReportUnexpectedException(Exception exception, CancellationToken ca
             // as any observation of ConnectionLostException indicates a bug (e.g. https://github.com/microsoft/vs-streamjsonrpc/issues/549).
             if (exception is ConnectionLostException)
             {
-                ClientDisconnectedSource.Cancel();
-
                 return true;
             }
 
             // Indicates bug on client side or in serialization, report NFW and propagate the exception.
             return FatalError.ReportWithoutCrashAndPropagate(exception);
         }
 
-        private static Exception OnUnexpectedException(CancellationToken cancellationToken)
+        private static Exception OnUnexpectedException(Exception exception, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            // If this is hit the cancellation token passed to the service implementation did not use the correct token.
+            if (exception is ConnectionLostException)
```

Comment:
this is probably worth documenting why we're doing it this way.  doesn't need to be in this PR though.

Reply:
It's documented a few lines above, and still under investigation.

https://github.com/dotnet/roslyn/blob/4123b1c06ee17b42644931995ab00b15907cd8c6/src/Workspaces/Remote/Core/RemoteCallback.cs#L103-L112